gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -133,6 +133,8 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
         public static final Pattern SKIP_CANCEL_STMTS = Pattern.compile(
             "^\\s*(DROP|TRUNCATE)\\s+(TABLE|INDEX)\\s+", Pattern.CASE_INSENSITIVE);
+	public static final Pattern DELETE_STMT = Pattern.compile(
+			"^\\s*DELETE\\s+FROM\\s+.*", Pattern.CASE_INSENSITIVE);
         private static final Pattern ignoreStmts = Pattern.compile(
```

Comment:
Looks like one of the memory checks is before parsing in prepMinion, for all others, there are QueryTreeNode is available, so probably those can use the QueryTreeNode.  Probably we may need to add  the checkMemory calls after parsing is done only?